src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -508,5 +509,21 @@ public static HashSet<string> GetLanguageWorkerListToStartInPlaceholder(this IEn
             }
             return placeholderRuntimeSet;
         }
+
+        public static bool IsApplicationInsightsAgentEnabled(this IEnvironment environment)
+        {
+            // cache the value of the environment variable
+            if (isApplicationInsightsAgentEnabled.HasValue)
```

Comment:
The only downside of having the logic this way would be if this value is enabled during placeholder mode and customer choses to disable it explicitly. They won't be able to.